src/main/kotlin/uk/gov/gdx/datashare/service/DataProcessor.kt

```diff
@@ -23,37 +24,23 @@ class DataProcessor(
     val dataProcessorMessage: DataProcessorMessage = objectMapper.readValue(message, DataProcessorMessage::class.java)
     log.info("Received event [{}] from [{}]", dataProcessorMessage.eventTypeId, dataProcessorMessage.publisher)
 
-    // lookup provider
-    val details = getDataFromProvider(dataProcessorMessage)
+    val dataId = dataProcessorMessage.id ?: UUID.randomUUID().toString()
```

Comment:
Probably a wider conversation, but is this valuable over just making this nullable?

Reply:
Yeah I think there's a wider conversation around what situation we don't have an id in - because we can't fetch data from the lev without the id - and in fact maybe the random uuid just hides the fact we don't have one in the db?